src/js/tech/ChromecastTech.js

```diff
@@ -0,0 +1,372 @@
+'use strict';
+
+var ChromecastSessionManager = require('../chromecast/ChromecastSessionManager'),
+    ChromecastTechUI = require('./ChromecastTechUI'),
+    _ = require('underscore'),
+    SESSION_TIMEOUT = 10 * 1000, // milliseconds
+    ChromecastTech;
+
+
+ChromecastTech = {
+   constructor: function(options) {
+      var subclass;
+
+      this._eventListeners = [];
+
+      this.videojsPlayer = this.videojs(options.playerId);
+      this._chromecastSessionManager = this.videojsPlayer.chromecastSessionManager;
+
+      // We have to initialize the UI here, before calling super.constructor
+      // because the constructor calls `createEl`, which references `this._ui`.
+      this._ui = new ChromecastTechUI();
+      this._ui.updatePoster(this.videojsPlayer.poster());
+
+      // Call the super class' constructor function
+      subclass = this.constructor.super_.apply(this, arguments);
+
+      this._remotePlayer = this._chromecastSessionManager.getRemotePlayer();
+      this._remotePlayerController = this._chromecastSessionManager.getRemotePlayerController();
+      this._listenToPlayerControllerEvents();
+      this.on('dispose', this._removeAllEventListeners.bind(this));
+
+      this._hasPlayedAnyItem = false;
+      this._requestTitle = options.requestTitleFn || _.noop;
+      this._requestSubtitle = options.requestSubtitleFn || _.noop;
+      // See `currentTime` function
+      this._initialStartTime = options.startTime || 0;
+
+      this._playSource(options.source, this._initialStartTime);
+      this.ready(function() {
+         this.setMuted(options.muted);
+      }.bind(this));
+
+      return subclass;
+   },
+
+   createEl: function() {
+      return this._ui.getDOMElement();
+   },
+
+   play: function() {
+      if (!this.paused()) {
+         return;
+      }
+      if (this.ended() && !this._isMediaLoading) {
+         // Restart the current item from the beginning
+         this._playSource({ src: this.videojsPlayer.src() }, 0);
+      } else {
+         this._remotePlayerController.playOrPause();
+      }
+   },
+
+   pause: function() {
+      if (!this.paused() && this._remotePlayer.canPause) {
+         this._remotePlayerController.playOrPause();
+      }
+   },
+
+   paused: function() {
+      return this._remotePlayer.isPaused || this.ended();
+   },
+
+   setSource: function(source) {
+      if (this._currentSource && this._currentSource.src === source.src && this._currentSource.type === source.type) {
```

Comment:
Can we add a comment why we need to track `this._currentSource` and don't rely on `this.videojsPlayer.src()`?